### PR TITLE
[APP] GrampsWeb

### PR DESCRIPTION
**Pull request by Greg0**
# Description

Gramps Web is a web app for collaborative genealogy. It is based on and interoperable with Gramps, the leading open source genealogy desktop application. Gramps Web is free & open source software and puts your privacy and your control of your research data first.

# Links

https://www.grampsweb.org/

## Summary by CodeRabbit

- **New Features**
  - Introduced a `docker-compose.yml` configuration for the `grampsweb` service, streamlining deployment with settings for image, ports, environment variables, volumes, networks, and Traefik labels for routing and SSL termination.
  
- **Configuration Updates**
  - Updated `config.json` to include comprehensive details such as app name, port, version, description, author information, source repository, and supported architectures.

### CODE REVIEW

File: apps/grampsweb/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+version: "3.7"
+
+services:
+  grampsweb:
+    image: ghcr.io/gramps-project/grampsweb:latest
```
Comment (JigSawFr) [inline apps/grampsweb/docker-compose.yml:5]: should never be latest, please use version of config.json

**Comment by JigSawFr** [comment]
Container name is missing and network too. Traefik should also be added.

**Review by JigSawFr — CHANGES_REQUESTED** [review]

**Comment by Greg0** [comment]
> Container name is missing and network too. Traefik should also be added.

Please verify if changes are correct

**Comment by Greg0** [comment]
@JigSawFr please verify changes if you'll find space for that

**Comment by JigSawFr** [comment]
@Greg0 please change the rest et policy to unless-stopped and it should be fine ;)

**Comment by Greg0** [comment]
> @Greg0 please change the restart policy to unless-stopped and it should be fine ;)

@JigSawFr Done

File: apps/grampsweb/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - ${APP_DATA_DIR}/users:/app/users # persist user database
+      - ${APP_DATA_DIR}/index:/app/indexdir # persist search index
+      - ${APP_DATA_DIR}/thumb_cache:/app/thumbnail_cache # persist thumbnails
+      - ${APP_DATA_DIR}/cache:/app/cache # persist export and report caches
+      - ${APP_DATA_DIR}/secret:/app/secret # persist flask secret
+      - ${APP_DATA_DIR}/db:/root/.gramps/grampsdb # persist Gramps database
+      - ${APP_DATA_DIR}/media:/app/media # persist media files
+      - ${APP_DATA_DIR}/tmp:/tmp
+    networks:
+      - tipi_main_network
```
[the remark this answers is not in the document]
Comment (JigSawFr) [reply apps/grampsweb/docker-compose.yml:23]: @Greg0 agree here, should be:
```
networks:
      - tipi_main_network
```

File: apps/grampsweb/config.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "../schema.json",
+  "name": "Gramps Web",
+  "available": true,
+  "exposable": true,
+  "port": 5000,
+  "id": "grampsweb",
+  "tipi_version": 3,
+  "version": "4.1.2",
```
Comment (JigSawFr) [inline apps/grampsweb/config.json:9]: Should correspond to the docker image version, so:
`v24.5.0`

File: apps/grampsweb/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+services:
+  grampsweb:
+    image: ghcr.io/gramps-project/grampsweb:v24.5.0
+    container_name: grampsweb
+    restart: unless-stopped
+    ports:
+      - ${APP_PORT}:5000 # host:docker
+    environment:
+      GRAMPSWEB_TREE: ${GRAMPS_TREE} # will create a new tree if not exists
```
Comment (JigSawFr) [inline apps/grampsweb/docker-compose.yml:11]: Why do you use a VAR env if not used in the config.json file ? So will be empty forever
Your var is referencing a non-existant var atm.

**Review by JigSawFr — CHANGES_REQUESTED** [review]

File: apps/grampsweb/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      # Local domain
+      traefik.http.routers.grampsweb-local-insecure.rule: Host(`grampsweb.${LOCAL_DOMAIN}`)
+      traefik.http.routers.grampsweb-local-insecure.entrypoints: web
+      traefik.http.routers.grampsweb-local-insecure.service: grampsweb
+      traefik.http.routers.grampsweb-local-insecure.middlewares: grampsweb-web-redirect
+      # Local domain secure
+      traefik.http.routers.grampsweb-local.rule: Host(`grampsweb.${LOCAL_DOMAIN}`)
+      traefik.http.routers.grampsweb-local.entrypoints: websecure
+      traefik.http.routers.grampsweb-local.service: grampsweb
+      traefik.http.routers.grampsweb-local.tls: true
```
Comment (JigSawFr) [inline apps/grampsweb/docker-compose.yml:48]: The following label is also missing and mandatory:
`runtipi.managed=true`

**Review by JigSawFr — CHANGES_REQUESTED** [review]

File: apps/grampsweb/docker-compose.yml (hunk 4 above)
Comment (Greg0) [reply apps/grampsweb/docker-compose.yml]: @JigSawFr I think now is correct. Check please

**Review by JigSawFr — APPROVED** [review]

**Pull request merged**